showcase/screens/Onboarding2.tsx

```diff
@@ -16,18 +16,58 @@ const pages = [
       'You did it! Great job at swiping that page!\nNow, try clicking on the CTA to see yet another example of an onboarding component.\nAlso, did you see how images up there take the whole width of your device, while dynamically being resized according to their height? Yes? Gr8!',
     image: require('../assets/onboarding.png'),
   },
+  {
+    title: "Look at me, I'm a title spanning two lines. Spiffy, uh?",
+    content:
+      "Oh hai! I'm a 2-page onboarding component. Try swiping to the next page, see what happens.",
+    image: require('../assets/onboarding2.png'),
+  },
+  {
+    title: 'title label',
+    content:
+      'You did it! Great job at swiping that page!\nNow, try clicking on the CTA to see yet another example of an onboarding component.\nAlso, did you see how images up there take the whole width of your device, while dynamically being resized according to their height? Yes? Gr8!',
+    image: require('../assets/onboarding.png'),
+  },
 ];
 
-class Onboarding extends React.PureComponent<OnboardingProps> {
+type OnboardingScreenState = {
+  index: number;
+};
+
+class Onboarding extends React.PureComponent<OnboardingProps, OnboardingScreenState> {
+  private onboardingComponent = createRef<OnboardingComponent>();
+  constructor(props: OnboardingProps) {
+    super(props);
+    this.state = { index: 0 };
+    this.onboardingComponent = createRef();
+  }
+
+  next() {
+    const { index } = this.state;
+    const endReached = index === pages.length - 1;
+    this.onboardingComponent?.current?.scrollPage(1, index + 2);
+    this.setState({ index: index + 1 });
```

Comment:
the 2 lines above should be in the `else` for `if (endReached)`. It looks like it's nitpicking, but if you try the page on the showcase app and go back from the 3rd Onboarding example back to the 2nd (the one that uses this function) you'll see that the component is left in an inconsistent state: it's at the last page, but the dot is not highlighted (because `index` is `4`)

Reply:
You are right also here, commit 7018631 thanks a lot.

Gif of the functionalities after the fix
<img src="https://user-images.githubusercontent.com/24992535/92082918-82fa3f80-edc5-11ea-9f78-d140970511a2.gif" width="250" />